core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -10435,32 +10435,30 @@ private void checkCustomColumnResolving(String table) {
         + "size => interval '1' hour))")
         .fails("Param 'data' not found in function 'SESSION'; did you mean 'DATA'\\?");
     sql("select * from table(\n"
-        + "^session(\n"
+        + "session(\n"
         + "data => table orders,\n"
         + "key => descriptor(productid),\n"
```

Comment:
I found that optional named argument is not working here, even though I have do 

```
    @Override public boolean isOptional(int i) {
      return false;
      return i == 2;
    }
```
I expect Calcite can detect required `timecol` has missing. I haven't figured out the reason of this.